service/core.go

```diff
@@ -164,7 +167,7 @@ func (c *Core) addAPIServerRoutes() {
 	c.authhandle("/pool/{pool}/branch/{branch}/merge/{child}", handleBranchMerge).Methods("POST")
 	c.authhandle("/pool/{pool}/branch/{branch}/revert/{commit}", handleRevertPost).Methods("POST")
 	c.authhandle("/pool/{pool}/stats", handlePoolStats).Methods("GET")
-	c.authhandle("/query", handleQuery).Methods("POST")
+	c.authhandle("/query", handleQuery).Methods("POST", "OPTIONS")
```

Comment:
I believe the cors middleware will handle this, so the OPTIONS method isn't required no?

Reply:
It is required because the router won't run the middleware unless there is a matching route (including method)

Reply:
```suggestion
	c.authhandle("/query", handleQuery).Methods("OPTIONS", "POST")
```

Reply:
Frustrating! But makes sense.